kernel/examples/write-table/src/main.rs

```diff
@@ -0,0 +1,322 @@
+            if parts.len() != 2 {
+                return Err(delta_kernel::Error::generic(format!(
+                    "Invalid field specification: {field}. Expected format: field_name:data_type"
+                )));
+            }
+
+            let (name, data_type) = (parts[0].trim(), parts[1].trim());
+            let data_type = match data_type {
+                "string" => DataType::STRING,
+                "integer" => DataType::INTEGER,
+                "long" => DataType::LONG,
+                "double" => DataType::DOUBLE,
+                "boolean" => DataType::BOOLEAN,
+                "timestamp" => DataType::TIMESTAMP,
+                _ => {
+                    return Err(delta_kernel::Error::generic(format!(
+                        "Unsupported data type: {data_type}"
+                    )));
+                }
+            };
+
+            Ok(StructField::nullable(name, data_type))
+        })
+        .collect::<DeltaResult<Vec<_>>>()?;
+
+    Ok(Arc::new(StructType::new(fields)))
+}
+
+/// Create a new Delta table with the given schema
+/// Creating a Delta table is not officially supported by kernel-rs yet, so we manually create the initial transaction log
+async fn create_table(
+    store: &Arc<dyn ObjectStore>,
+    table_url: &Url,
+    schema: &SchemaRef,
+) -> DeltaResult<()> {
+    let table_id = "dummy_table";
+    let schema_str = serde_json::to_string(&schema)?;
+
+    let (reader_features, writer_features) = {
+        let reader_features: Vec<&'static str> = vec![];
+        let writer_features: Vec<&'static str> = vec![];
+
+        // TODO: Support adding specific table features
+        (reader_features, writer_features)
+    };
+
+    let protocol = json!({
+        "protocol": {
+            "minReaderVersion": 3,
+            "minWriterVersion": 7,
+            "readerFeatures": reader_features,
+            "writerFeatures": writer_features,
+        }
+    });
+    let partition_columns: Vec<String> = vec![];
+    let metadata = json!({
+        "metaData": {
+            "id": table_id,
+            "format": {
+                "provider": "parquet",
+                "options": {}
+            },
+            "schemaString": schema_str,
+            "partitionColumns": partition_columns,
+            "configuration": {},
+            "createdTime": 1677811175819u64
+        }
+    });
+
+    let data = [
+        to_vec(&protocol).unwrap(),
+        b"\n".to_vec(),
+        to_vec(&metadata).unwrap(),
+    ]
+    .concat();
+
+    // Write the initial transaction with with protocol and metadata to 0.json
+    let path = table_url.join("_delta_log/00000000000000000000.json")?;
+    store
+        .put(&Path::from_url_path(path.path())?, data.into())
+        .await?;
+
+    println!("✓ Created Delta table with schema: {schema:#?}");
+    Ok(())
+}
+
+/// Create sample data based on the schema
+fn create_sample_data(schema: &SchemaRef, num_rows: usize) -> DeltaResult<ArrowEngineData> {
```

Comment:
it feels like this PR may get _simpler_ if we allow for writing arbitrary JSON data? we could just delete all of this

Reply:
Resolving this to consolidate the discussion around JSON input.

[See here](https://github.com/delta-io/delta-kernel-rs/pull/1119#discussion_r2256673507)